src/error.rs

```diff
@@ -0,0 +1,50 @@
+use thiserror::Error;
```

Comment:
For this library we have this turned on  #![deny(missing_docs)].

Once this gets public, it will require a documentation of this module.

Reply:
For now I have written some basic documentation. I think that should be sufficient for now.